src/Arcus.Security.Providers.AzureKeyVault/Authentication/CertificateBasedAuthentication.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using Arcus.Security.Providers.AzureKeyVault.Authentication.Interfaces;
+using GuardNet;
+using Microsoft.Azure.KeyVault;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+
+namespace Arcus.Security.Providers.AzureKeyVault.Authentication
+{
+    /// <summary>
+    ///     Azure Key Vault <see cref="IKeyVaultAuthentication"/> by using client ID and certificate to authenticate the <see cref="IKeyVaultClient"/>.
+    /// </summary>
+    public class CertificateBasedAuthentication : IKeyVaultAuthentication
+    {
+        private readonly string _applicationId;
+        private readonly X509Certificate2 _certificate;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CertificateBasedAuthentication"/> class.
+        /// </summary>
+        /// <param name="applicationId">The identifier of the application requesting the authentication token.</param>
+        /// <param name="certificate">The certificate that is used as credential.</param>
+        public CertificateBasedAuthentication(string applicationId, X509Certificate2 certificate)
```

Comment:
We use `clientId` in the other authentication option, can you please align?
https://github.com/arcus-azure/arcus.security/blob/master/src/Arcus.Security.Providers.AzureKeyVault/Authentication/ServicePrincipalAuthenticator.cs#L25

Reply:
Azure docs talk about Application ID; but OK to allgin.

Reply:
Then align the other part ;)

Reply:
:) OK

Reply:
That's a fair point, tracking this in #61 to align this